specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1409 @@
+    /// is implicitly unpinned when the process terminates.
+    Process,
+
+    /// The lifetime artifact is an absolute filename or path.
+    /// The package dependency is implicitly unpinned when this is deleted.
+    FilePath,
+
+    /// The lifetime artifact is a registry key in the format
+    /// 'root\\subkey' where root is one of the following: HKLM, HKCU, HKCR, HKU.
+    /// The package dependency is implicitly unpinned when this is deleted.
+    RegistryKey,
+};
+
+/// Options when 'pinning' a package dependency
+runtimeclass PinPackageDependencyOptions
+{
+    PinPackageDependencyOptions();
+
+    /// Optional filtering by cpu architecture(s)
+    PackageDependencyProcessorArchitectures Architectures;
+
+    /// Do not verify at least 1 matching package exists when pinning a package dependency
+    Boolean VerifyDependencyResolution = true;
+
+    /// The kind of lifetime artifact for this package dependency.
+    PackageDependencyLifetimeArtifactKind LifetimeArtifactKind;
+
+    /// The lifetime artifact when pinning a package dependency. The value depends on the LifetimeArtifactKind value.
+    String LifetimeArtifact;
+}
+
+/// Options when adding a package dependency
+runtimeclass AddPackageDependencyOptions
+{
+    AddPackageDependencyOptions();
+
+    /// The rank when adding the package dependency to a a package graph.
+    /// @note A package graph is sorted in ascending order from -infinity...0...+infinity
+    /// @note The default value is zero
+    Int32 Rank;
+
+    /// If a package dependency resolves to a package with dependencies and this option is true
+    /// the dependencies are ignored; only the package is added to the package graph. By default
+    /// a resolved package and its dependencies are added to the package graph.
+    Boolean OnlyUseFirstPackageFamily;
+
+    /// If a package is added to a package graph with a package of the same rank (aka a collision on rank)
+    /// and this option is true the package is prepended to the set of packages of the same rank.
+    /// By default packages are appended to the set of packages with the same rank.
+    Boolean PrependIfRankCollision;
+}
+
+/// TBD
+runtimeclass PackageDependency
+{
+    /// Create an intstance of the package dependency identified by id.
+    PackageDependency(String id);
+
+    /// Return the package dependency id.
+    String Id { get; }
+
+    /// Return true if the package dependencies would produce the same
+    /// package when resolved e.g. whether they share the same packageFamilyName,
+    /// minVersion, and packageDependencyProcessorArchitectures values.
+    static Boolean AreEquivalent(String packageDependencyId1, String packageDependencyId2);
+
+    /// Return true if this and other would produce the same
+    /// package when resolved e.g. whether they share the same packageFamilyName,
+    /// minVersion, and packageDependencyProcessorArchitectures values.
+    Boolean IsEquivalent(PackageDependency other);
```

Comment:
There's precedence for Equals: [Uri.Equals](https://docs.microsoft.com/uwp/api/Windows.Foundation.Uri.Equals)